vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -1,1726 +1,1548 @@
+	}
+	if os.Getenv("VSPHERE_DATASTORE") == "" {
+		t.Skip("set VSPHERE_DATASTORE to run vsphere_virtual_machine acceptance tests")
+	}
+	if os.Getenv("VSPHERE_TEMPLATE") == "" {
+		t.Skip("set VSPHERE_TEMPLATE to run vsphere_virtual_machine acceptance tests")
+	}
+}
 
-	if v := os.Getenv("VSPHERE_TEMPLATE"); v == "" {
-		t.Fatal("env variable VSPHERE_TEMPLATE must be set for acceptance tests")
+func testAccResourceVSphereVirtualMachineCheckExists(expected bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		_, err := testGetVirtualMachine(s, "vm")
+		if err != nil {
+			if ok, _ := regexp.MatchString("virtual machine with UUID \"[-a-f0-9]+\" not found", err.Error()); ok && !expected {
+				// Expected missing
+				return nil
+			}
+			return err
+		}
+		if !expected {
+			return errors.New("expected VM to be missing")
+		}
+		return nil
 	}
+}
 
-	if v := os.Getenv("VSPHERE_IPV4_GATEWAY"); v == "" {
-		t.Fatal("env variable VSPHERE_IPV4_GATEWAY must be set for acceptance tests")
+// testAccResourceVSphereVirtualMachineCheckPowerState is a check to check for
+// a VirtualMachine's power state.
+func testAccResourceVSphereVirtualMachineCheckPowerState(expected types.VirtualMachinePowerState) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		props, err := testGetVirtualMachineProperties(s, "vm")
+		if err != nil {
+			return err
+		}
+		actual := props.Runtime.PowerState
+		if expected != actual {
+			return fmt.Errorf("expected power state to be %s, got %s", expected, actual)
+		}
+		return nil
 	}
+}
 
-	if v := os.Getenv("VSPHERE_IPV4_ADDRESS"); v == "" {
-		t.Fatal("env variable VSPHERE_IPV4_ADDRESS must be set for acceptance tests")
+// testAccResourceVSphereVirtualMachineCheckHostname is a check to check for a
+// VirtualMachine's hostname. The check uses guest info, so VMware tools needs
+// to be installed.
+func testAccResourceVSphereVirtualMachineCheckHostname(expected string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		props, err := testGetVirtualMachineProperties(s, "vm")
+		if err != nil {
+			return err
+		}
+		actual := props.Guest.HostName
+		if expected != actual {
+			return fmt.Errorf("expected hostname to be %s, got %s", expected, actual)
+		}
+		return nil
 	}
+}
+
+// testAccResourceVSphereVirtualMachineCheckExtraDisks is a check for proper
+// parameters on the vsphere_virtual_machine extra disks test. This is a very
+// specific check that checks for the specific disk devices and respective
+// backings, and expects them in the exact order outlined in the function.
+func testAccResourceVSphereVirtualMachineCheckExtraDisks() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		props, err := testGetVirtualMachineProperties(s, "vm")
+		if err != nil {
+			return err
+		}
+
+		var foundEager, foundLazy, foundThin bool
+		expectedEagerName := "terraform-test-extra-eager.vmdk"
```

Comment:
Is there any chance for these names to change? I'm thinking we make these constants